price-feeder/oracle/oracle.go

```diff
@@ -259,74 +244,92 @@ func (o *Oracle) SetPrices(ctx context.Context, acceptList oracletypes.DenomList
 		o.logger.Debug().Err(err).Msg("failed to get ticker prices from provider")
 	}
 
-	filteredCandles, err := o.filterCandleDeviations(providerCandles)
+	computedPrices, err := GetComputedPrices(o.logger, providerCandles, providerPrices)
 	if err != nil {
 		return err
 	}
 
+	if len(computedPrices) != len(requiredRates) {
+		return fmt.Errorf("unable to get prices for all exchange candles")
+	}
+	for base := range requiredRates {
+		if _, ok := computedPrices[base]; !ok {
+			return fmt.Errorf("reported prices were not equal to required rates, missed: %s", base)
+		}
+	}
+
+	o.prices = computedPrices
+	return nil
+}
+
+// GetComputedPrices gets the candle and ticker prices and compute it.
+// it returns candles TVWAP if possible, if not possible (not available
+// or due to some staleness) it will use the most recent ticker prices
+// e VWAP instead.
```

Comment:
```suggestion
// GetComputedPrices gets the candle and ticker prices and computes it.
// It returns candles' TVWAP if possible, if not possible (not available
// or due to some staleness) it will use the most recent ticker prices
// and the VWAP formula instead.
```